clients/vault/src/system.rs

```diff
@@ -940,4 +934,12 @@ impl VaultService {
 		tracing::info!("Got new block at height {startup_height}");
 		Ok(startup_height)
 	}
+
+	/// shuts down the resubmission task running in the background
+	async fn shutdown_wallet(&self) {
```

Comment:
Let's rename to `try_shutdown_wallet()` because I suppose this will only try to shut it down but not error if it's not running (yet) (?)